compact/hcal/hcal_barrel_gdml.xml

```diff
@@ -0,0 +1,7682 @@
+        <position name="OuterHCalTile05_EJ200_v108" unit="mm" x="488.6627" y="803.9563" z="7.0"/>
+        <position name="OuterHCalTile05_EJ200_v109" unit="mm" x="488.7138" y="804.3445" z="1.5"/>
+        <position name="OuterHCalTile05_EJ200_v110" unit="mm" x="488.7138" y="804.3445" z="7.0"/>
+        <position name="OuterHCalTile05_EJ200_v111" unit="mm" x="488.6627" y="804.7327" z="1.5"/>
+        <position name="OuterHCalTile05_EJ200_v112" unit="mm" x="488.6627" y="804.7327" z="7.0"/>
+        <position name="OuterHCalTile05_EJ200_v113" unit="mm" x="488.5128" y="805.0945" z="1.5"/>
+        <position name="OuterHCalTile05_EJ200_v114" unit="mm" x="488.5128" y="805.0945" z="7.0"/>
+        <position name="OuterHCalTile05_EJ200_v115" unit="mm" x="488.2745" y="805.4052" z="1.5"/>
+        <position name="OuterHCalTile05_EJ200_v116" unit="mm" x="488.2745" y="805.4052" z="7.0"/>
+        <position name="OuterHCalTile05_EJ200_v117" unit="mm" x="487.9638" y="805.6435" z="1.5"/>
+        <position name="OuterHCalTile05_EJ200_v118" unit="mm" x="487.9638" y="805.6435" z="7.0"/>
+        <position name="OuterHCalTile05_EJ200_v119" unit="mm" x="487.602" y="805.7934" z="1.5"/>
+        <position name="OuterHCalTile05_EJ200_v120" unit="mm" x="487.602" y="805.7934" z="7.0"/>
+        <position name="OuterHCalTile05_EJ200_v121" unit="mm" x="485.2972" y="798.9898" z="1.5"/>
+        <position name="OuterHCalTile05_EJ200_v122" unit="mm" x="485.2972" y="798.9898" z="7.0"/>
+        <position name="OuterHCalTile05_EJ200_v123" unit="mm" x="485.5992" y="798.754" z="1.5"/>
+        <position name="OuterHCalTile05_EJ200_v124" unit="mm" x="485.5992" y="798.754" z="7.0"/>
+        <position name="OuterHCalTile05_EJ200_v125" unit="mm" x="485.9511" y="798.6024" z="1.5"/>
+        <position name="OuterHCalTile05_EJ200_v126" unit="mm" x="485.9511" y="798.6024" z="7.0"/>
+        <position name="OuterHCalTile05_EJ200_v127" unit="mm" x="486.33" y="798.5449" z="1.5"/>
+        <position name="OuterHCalTile05_EJ200_v128" unit="mm" x="486.33" y="798.5449" z="7.0"/>
+        <position name="OuterHCalTile05_EJ200_v129" unit="mm" x="486.7111" y="798.5853" z="1.5"/>
+        <position name="OuterHCalTile05_EJ200_v130" unit="mm" x="486.7111" y="798.5853" z="7.0"/>
+        <position name="OuterHCalTile05_EJ200_v131" unit="mm" x="487.0695" y="798.7209" z="1.5"/>
+        <position name="OuterHCalTile05_EJ200_v132" unit="mm" x="487.0695" y="798.7209" z="7.0"/>
+        <position name="OuterHCalTile05_EJ200_v133" unit="mm" x="487.3818" y="798.9429" z="1.5"/>
+        <position name="OuterHCalTile05_EJ200_v134" unit="mm" x="487.3818" y="798.9429" z="7.0"/>
+        <position name="OuterHCalTile05_EJ200_v135" unit="mm" x="487.6277" y="799.2367" z="1.5"/>
+        <position name="OuterHCalTile05_EJ200_v136" unit="mm" x="487.6277" y="799.2367" z="7.0"/>
+        <position name="OuterHCalTile05_EJ200
```

Comment:
The readouts section is intended, e.g., to implement very large numbers of fine pixels in an otherwise homogeneous material. The geometry would be slow as molasses in Winnipeg if we were to implement those pixels in geometry. So, the volume is segmented 'outside' of the geometry. DD4hep makes the single volume sensitive, then figures out from the local hit position which pixel was hit, and assigns that as a 64-bit cell ID. Each 64-bit cell ID is unique, since it also stores the detector ID (`system`) and other components such as `sector`, `tower`, etc.

If you don't need the finer segmentation than what is provided at the volume level, you can use the `NoSegmentation` type, like the backward ecal (`compact/ecal_backward_PbWO4.xml`):
```xml
    <readout name="EcalEndcapNHits">
      <segmentation type="NoSegmentation" key="sector"/>
      <id>system:8,sector:4,module:20</id>
    </readout>
```
In this case, there is no need for the `x` and `y` fields since there is no XY segmentation at the tower level.

There are signal merging reconstruction algorithms in case the signals from multiple towers are combined in hardware before digitization (multiple fibers to the same PMT is the usual example).

Reply:
Changing the readout name to HcalBarrelHits gives a runtime error: 

`runtime_error: dd4hep: Attempt to add an already existing object:HcalBarrelHits.`

in the "linux-lcg / build-lcg-ubuntu-2004 (LCG_101/x86_64-ubuntu2004-gcc9-opt) (pull_request) " check.  Presumably a conflict with the same readout name in the old Hcal code?

Reply:
Had both versions of the HCAL in place due to messed up jinja template.  FIxed and committed.